src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -486,6 +486,11 @@ trait OptimisticTransactionImpl extends TransactionalWrite with SQLMetricsReport
       } else {
         changedDataAddedFiles ++= winningCommitActions.collect { case a: AddFile => a }
       }
+      val rearrangeOnly =
```

Comment:
You might consider turning both checks into functions, they are used 1-2 more times throughout the file.